src/profile/supporting_types/host_info.rs

```diff
@@ -0,0 +1,125 @@
+use crate::prelude::*;
+
+#[derive(
+    Clone, Debug, PartialEq, Eq, Hash, derive_more::Display, uniffi::Record,
+)]
+#[display("Host {} '{} {}' running on {}, firmware: {}", id, description.name, description.model, host_os_version, host_app_version)]
+pub struct HostInfo {
+    /// A best effort stable and unique identifier of this
+    /// device.
+    pub id: DeviceID,
+
+    /// The date this description of the device was made, might
+    /// be equal to when the app was first ever launched on the
+    /// device.
+    pub date: Timestamp,
+
+    /// A short description of the device, we devices should
+    /// read the device model and a given name from the device
+    /// if they are able to.
+    pub description: DeviceInfoDescription,
+
+    /// The **current** version of the device's operating system, e.g. "iOS 17.4.1".
+    pub host_os_version: String,
+
+    /// The **current** version of the host app, for example the Radix iOS Wallet version - e.g. "1.6.1"
+    pub host_app_version: String,
+
+    /// The vendor of the host client, e.g. "Apple" for iPhone clients,
+    /// or "Samsung" for Android clients.
+    pub host_vendor: String,
+}
+
+// HostInfo can be converted to DeviceInfo but not the opposite
+// without contacting HostInfoDriver
+#[allow(clippy::from_over_into)]
+impl Into<DeviceInfo> for HostInfo {
+    fn into(self) -> DeviceInfo {
+        let info = self.clone();
+        DeviceInfo::new(
+            info.id,
+            info.date,
+            info.description,
+            info.host_os_version,
+            info.host_app_version,
+            info.host_vendor,
+        )
+    }
+}
+
+impl HasSampleValues for HostInfo {
+    fn sample() -> Self {
+        Self {
+            id: DeviceID::from_str("66F07CA2-A9D9-49E5-8152-77ACA3D1DD74")
+                .unwrap(),
+            date: Timestamp::sample(),
+            description: DeviceInfoDescription {
```

Comment:
I would expected DeviceInfoDescription to also implement HasSampleValues

Reply:
It does but does not follow the logic of one sample being for an iOS client and other for Android.